### PR TITLE
Issue #41: Calculate delegation rewards for nodes/delegators

### PR DESCRIPTION
**Pull request by rattrap**

### CODE REVIEW

**Review by farebord — APPROVED** [review]
Looks good to me! There can be optimizations regarding list management, but it won't increase performance that much. 👍

File: services/delegation/src/modules/reward/reward.service.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    private nodeCommissionRepository: Repository<NodeCommission>,
+    @InjectRepository(Delegation)
+    private delegationRepository: Repository<Delegation>,
+    private config: ConfigService,
+  ) {
+    this.eligibilityThreshold = this.config.get<number>(
+      'delegation.eligibilityThreshold',
+    );
+  }
+  async getRewards(at = moment().utc().unix()) {
```
Comment (VargaElod23) [inline services/delegation/src/modules/reward/reward.service.ts:42]: Return type would be good for type chaining 👍🏼

File: services/delegation/src/modules/reward/reward.service.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        accountBalances.set(user, (accountBalances.get(user) || 0) + balance);
+      });
+    }
+
+    return {
+      nodeBalances,
+      accountBalances,
+    };
+  }
+  async getRewardsForNode(node: Node, at = moment().utc().unix()) {
```
Comment (VargaElod23) [inline services/delegation/src/modules/reward/reward.service.ts:68]: Return type would be good for type chaining 👍🏼

File: services/delegation/src/modules/reward/reward.service.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+    const getCurrentDelegations = (
+      startsAt: number,
+      endsAt: number,
+    ): IntervalValue[] => {
+      return [...delegationIntervalTree.search(new Interval(startsAt, endsAt))];
+    };
+
+    const getTotalDelegationAmount = (delegations: IntervalValue[]): number => {
+      return delegations.reduce((acc: number, cur: IntervalValue) => {
```
Comment (VargaElod23) [inline services/delegation/src/modules/reward/reward.service.ts:100]: Nice solution 🦖

**Review by VargaElod23 — APPROVED** [review]
PR approved, some explicit type specification would be nice but overall great job 🥇

**Pull request merged**